platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/TestMethodResult.java

```diff
@@ -31,12 +31,12 @@ public class TestMethodResult {
     private final TestResult.ResultType resultType;
     private final long duration;
     private final long endTime;
-    private final List<SerializedMetadata> metadatas;
+    private final List<TestMetadataEvent> metadatas;
     private final List<SerializableFailure> failures;
```

Comment:
🤔 I'm unsure of re-using `TestEventMetadata` to replace `SerializedMetadata`.

It seems fine _now_, but it also:
- Restricts how we can evolve this class in the future, since now it must be directly serializable.
- Seems misaligns with how we treat failures here (which makes sense, failures are more complex and this is a simple "value type").